internal/handlers/publishers.go

```diff
@@ -79,96 +86,132 @@ func (p *Publisher) GetPublisher(ctx *fiber.Ctx) error {
 
 // PostPublisher creates a new publisher.
 func (p *Publisher) PostPublisher(ctx *fiber.Ctx) error {
-	request := common.Publisher{}
+	normalize := normalizer.NewNormalizer()
+	request := new(common.Publisher)
 
-	if err := ctx.BodyParser(&request); err != nil {
-		return common.Error(fiber.StatusBadRequest, "can't create Publisher", "invalid json")
+	err := common.ValidateRequestEntity(ctx, request, "can't create Publisher")
+	if err != nil {
+		return err //nolint:wrapcheck
 	}
 
-	if err := common.ValidateStruct(&request); err != nil {
-		return common.ErrorWithValidationErrors(
-			fiber.StatusUnprocessableEntity, "can't create Publisher", "invalid format", err,
-		)
+	publisher := &models.Publisher{}
+
+	normalizedEmail := normalize.Normalize(request.Email)
+
+	publisher.ID = utils.UUIDv4()
+	publisher.Email = normalizedEmail
+
+	if request.ExternalCode != "" {
+		publisher.ExternalCode = request.ExternalCode
 	}
 
-	publisher := &models.Publisher{
-		ID:    utils.UUIDv4(),
-		Email: request.Email,
+	if request.Description != "" {
+		publisher.Description = request.Description
 	}
 
-	for _, URLAddress := range request.CodeHosting {
-		publisher.CodeHosting = append(publisher.CodeHosting, models.CodeHosting{ID: utils.UUIDv4(), URL: URLAddress.URL})
+	for _, codeHost := range request.CodeHosting {
+		urlAddress, _ := url.Parse(codeHost.URL)
+		normalizedURL := purell.NormalizeURL(urlAddress, normalizeFlags)
+
+		publisher.CodeHosting = append(publisher.CodeHosting, models.CodeHosting{ID: utils.UUIDv4(), URL: normalizedURL})
 	}
 
 	if err := p.db.Create(&publisher).Error; err != nil {
-		return common.Error(fiber.StatusInternalServerError, "can't create Publisher", "db error")
+		switch database.WrapErrors(err) { //nolint:errorlint
+		case common.ErrRecordNotFound:
+			return common.Error(fiber.StatusNotFound,
+				"can't create Publisher",
+				"Publisher was not found")
+		case common.ErrUniqueConstraint:
+			return common.Error(fiber.StatusConflict,
```

Comment:
Nice!

We need to document that 409 response in the OAS file.

Reply:
Right, updated.